File: docs/src/testers/sessions/capabilities/dev_environment.ts
// @sniptest filename=dev_environment.ts
// @sniptest show=1-11
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

// Only use live view in development
const isDev = process.env.ENV === 'development';

const session = client.Session({ open_viewer: isDev });
await session.use(async () => {
  await session.execute({ type: "goto", url: "https://example.com" });
});

const status = await session.status();

export { status };

File: docs/src/testers/sessions/capabilities/dev_environment.py
# @sniptest filename=dev_environment.py
# @sniptest show=1-11
import os

from notte_sdk import NotteClient

client = NotteClient()

# Only use live view in development
is_dev = os.getenv("ENV") == "development"

with client.Session(open_viewer=is_dev) as session:
    session.execute(type="goto", url="https://example.com")

status = session.status()
